src/device.cc

```diff
@@ -281,33 +282,117 @@ std::shared_ptr<SubHidrawConnection> SubHidrawConnection::create(const DeviceSca
 
   connection->m_details.phys = sd.phys;
   connection->disableWrite(); // disable write notifier
+  connection->initSubDevice();
   return connection;
 }
 
 // -------------------------------------------------------------------------------------------------
-ssize_t SubDeviceConnection::sendData(const QByteArray& hidppMsg)
+void SubDeviceConnection::initSubDevice()
+{
+  struct timespec ts;
+  int msec = 50;
+  ts.tv_sec = msec / 1000;
+  ts.tv_nsec = (msec % 1000) * 1000000;
```

Comment:
arbitrary waiting time (also: `ts.tv_sec` needs to be `0` in this case) - but see comment on `nanosleeps`

Reply:
nonosleep function is now removed in the latest commit.